pkg/exchange/binance/convert.go

```diff
@@ -90,6 +90,132 @@ func toGlobalMarginAccount(account *binance.MarginAccount) *types.MarginAccount
+		TotalMaintMargin:            fixedpoint.MustNewFromString(account.TotalMaintMargin),
+		TotalMarginBalance:          fixedpoint.MustNewFromString(account.TotalMarginBalance),
+		TotalOpenOrderInitialMargin: fixedpoint.MustNewFromString(account.TotalOpenOrderInitialMargin),
+		TotalPositionInitialMargin:  fixedpoint.MustNewFromString(account.TotalPositionInitialMargin),
+		TotalUnrealizedProfit:       fixedpoint.MustNewFromString(account.TotalUnrealizedProfit),
+		TotalWalletBalance:          fixedpoint.MustNewFromString(account.TotalWalletBalance),
+		UpdateTime:                  account.UpdateTime,
+	}
+}
+
+func toGlobalFuturesUserAssets(userAssets []*futures.AccountAsset) (retAssets []types.FuturesUserAsset) {
+	for _, asset := range userAssets {
+		retAssets = append(retAssets, types.FuturesUserAsset{
+			Asset:				asset.Asset,
+			InitialMargin:		fixedpoint.MustNewFromString(asset.InitialMargin),
+			MaintMargin:            fixedpoint.MustNewFromString(asset.MaintMargin),
+			MarginBalance:          fixedpoint.MustNewFromString(asset.MarginBalance),
+			MaxWithdrawAmount:      fixedpoint.MustNewFromString(asset.MaxWithdrawAmount),
+			OpenOrderInitialMargin: fixedpoint.MustNewFromString(asset.OpenOrderInitialMargin),
+			PositionInitialMargin:  fixedpoint.MustNewFromString(asset.PositionInitialMargin),
+			UnrealizedProfit:       fixedpoint.MustNewFromString(asset.UnrealizedProfit),
+			WalletBalance:         fixedpoint.MustNewFromString(asset.WalletBalance),
+		})
+	}
+
+	return retAssets
+}
+
+func toGlobalFuturesUserPositions(userPositions []*futures.AccountPosition) (retAssets []types.FuturesUserPosition) {
+	for _, position := range userPositions {
+		retAssets = append(retAssets, types.FuturesUserPosition{
+			Isolated:               position.Isolated,
+			Leverage:               fixedpoint.MustNewFromString(position.Leverage),
+			InitialMargin:          fixedpoint.MustNewFromString(position.InitialMargin),
+			MaintMargin:            fixedpoint.MustNewFromString(position.MaintMargin),
+			OpenOrderInitialMargin: fixedpoint.MustNewFromString(position.OpenOrderInitialMargin),
+			PositionInitialMargin:  fixedpoint.MustNewFromString(position.PositionInitialMargin),
+			Symbol:                 position.Symbol,
+			UnrealizedProfit:       fixedpoint.MustNewFromString(position.UnrealizedProfit),
+			EntryPrice:             fixedpoint.MustNewFromString(position.EntryPrice),
+			MaxNotional:            fixedpoint.MustNewFromString(position.MaxNotional),
+			PositionSide:           string(position.PositionSide),
+			PositionAmt:            fixedpoint.MustNewFromString(position.PositionAmt),
+			Notional:               fixedpoint.MustNewFromString(position.Notional),
+			IsolatedWallet:         fixedpoint.MustNewFromString(position.IsolatedWallet),
+			UpdateTime:             position.UpdateTime,
+		})
+	}
+
+	return retAssets
+}
+
+// Isolated
+func toGlobalIsolatedFuturesAccount(account *futures.Account) *types.IsolatedFuturesAccount {
```

Comment:
should be converted into types.Account (global type), but we can extend the future fields there

Reply:
you can add a Type field in the types.Account struct to indicate that it's a futures account.

for example,

```go
type AccountType string
const (
    AccountTypeFutures = AccountType("futures")
    AccountTypeSpot = AccountType("spot")
)
```